### PR TITLE
Stats Admin: Add Odyssey Widget behind feature flag

### LINKED ISSUE #29795: Stats Widget: Keeping the old widgets and boot up the new one behind a feature flag

### PR DESCRIPTION
**Pull request by kangzj**
Fixes #29795 

## Proposed changes:

The PR adds Odyssey Widget behind `odyssey_widget=1` for the WP Dashboard.

- Adds class `WP_Dashboard_Odyssey_Widget`
- Boot up Odyssey Widget when `odyssey_widget=1` exists in URL

### Other information:

- [ ] Have you written new tests for your changes, if applicable?
- [ ] Have you checked the E2E test CI results, and verified that your changes do not break them?
- [ ] Have you tested your changes on WordPress.com, if applicable (if so, you'll see a generated comment below with a script to run)?

## Jetpack product discussion
n/a

## Does this pull request change what data or activity we track or use?
no

## Testing instructions:

- Please follow instructions from https://github.com/Automattic/wp-calypso/pull/75190

### CODE REVIEW

File: projects/packages/stats-admin/src/class-odyssey-config-data.php [hunk 1]
```diff
@@ -53,6 +53,10 @@ public function get_data() {
 			'sections'                       => array(),
 			// Features are inlined @see https://github.com/Automattic/wp-calypso/pull/70122
 			'features'                       => array(),
+			// Intended for apps that do not use redux.
+			'gmt_offset'                     => $this->get_gmt_offset(),
+			// TODO: check wether this works with Pressable.
```
Comment (jsnmoon) [inline projects/packages/stats-admin/src/class-odyssey-config-data.php]:
```suggestion
			// TODO: check whether this works with Pressable.
```

😉

File: projects/packages/stats-admin/src/class-wp-dashboard-odyssey-widget.php [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+
+	/**
+	 * Load the admin scripts.
+	 */
+	public function load_admin_scripts() {
+		$config_data = ( new Odyssey_Config_Data() )->get_data();
+		// The widget doesn't use redux.
+		unset( $config_data['intial_state'] );
+		// TODO: change `configData` to a more unique name to avoid future conflicts.
+		( new Odyssey_Assets() )->load_admin_scripts( 'jetpack_stats_widget', 'widget-loader.min', 'configData', $config_data );
```
Comment (jsnmoon) [inline projects/packages/stats-admin/src/class-wp-dashboard-odyssey-widget.php:53]: Could we do this in this PR?

**Review by jsnmoon — DISMISSED** [review]
Works as described.

**Comment by jsnmoon** [comment]
I pushed a trunk merge into the branch.

File: projects/packages/stats-admin/src/class-wp-dashboard-odyssey-widget.php (hunk 2 above)
Comment (kangzj) [reply projects/packages/stats-admin/src/class-wp-dashboard-odyssey-widget.php:53]: The Odyssey side would need a bit of work as well and that's the reason I pushed it back - added an issue [here](https://github.com/Automattic/wp-calypso/issues/75257) to track it. Will definitely finish the change before it going live 😄

**Review by jsnmoon — APPROVED** [review]

File: projects/packages/stats-admin/src/class-wp-dashboard-odyssey-widget.php (hunk 2 above)
Comment (jsnmoon) [reply projects/packages/stats-admin/src/class-wp-dashboard-odyssey-widget.php:53]: Sounds good! 👍

**Pull request merged**